### PR TITLE
fix power cell slots not affected by emp

### LINKED ISSUE #16154: emp pulse doesn't affect batteries in powercell slots
## Description

**Reproduction**

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

closes #16154

**Media**

Peek 2023-05-06 11-20

**Changelog**

:cl:
- fix: fixed items with powercell slots not being affected by emp pulse

### CODE REVIEW

File: Content.Server/PowerCell/PowerCellSystem.cs [hunk 1]
```diff
@@ -303,4 +305,13 @@ private void OnBatteryExamined(EntityUid uid, BatteryComponent component, Examin
...
         args.PushMarkup(Loc.GetString("power-cell-component-examine-details", ("currentCharge", $"{charge:F0}")));
     }
+
+    private void OnEmpPulse(EntityUid uid, PowerCellSlotComponent component, ref EmpPulseEvent args)
+    {
+        if (TryGetBatteryFromSlot(uid, out var batteryUid, out var battery, component))
+        {
+            args.Affected = true;
+            _battery.UseCharge(batteryUid.Value, args.EnergyConsumption, battery);
+        }
```
Comment (metalgearsloth) [inline Content.Server/PowerCell/PowerCellSystem.cs:315]: Should this just relay the pulse event to the battery?

**Comment by deltanedas** [comment]
todo ninja real;

@Slava0135 does this have way to make slota immune to emp

**Comment by Slava0135** [comment]
it seems like items inside any container (but not inventory for some reason) are not affected

**Comment by deltanedas** [comment]
killer i think you might be able to iterate transform children
also please implement some way to make a powercell slot emp resistant i need for ninja

**Comment by Slava0135** [comment]
image
lookup doesn't check for nested containers (although its recursive?)

**Pull request closed without merging**